scipy/linalg/_matfuncs_expm.c

```diff
@@ -0,0 +1,2375 @@
+    if (m == 3)
+    {
+        // Am[2], Am[3] and Am[4] are free
+        b[0] = 120.0;
+        b[1] = 60.0;
+        b[2] = 12.0;
+        cb[0] = CPLX_C(120.0, 0.0);
+        cb[1] = CPLX_C(60.0, 0.0);
+
+        // U = Am[0] @ Am[1] + 60.*Am[0]
+        // V = 12.*Am[1] + 120*I_n
+
+         // Am[3] = Am[0]
+        ccopy_(&n2, Am, &int1, &Am[3*n2], &int1);
+        // Am[3] = Am[0] @ Am[1] + 60*Am[0]
+        cgemm_("N", "N", &n, &n, &n, &cdbl1, &Am[n2], &n, Am, &n, &cb[1], &Am[3*n2], &n);
+
+        // 12*Am[1]
+        csscal_(&n2, &b[2], &Am[n2], &int1);
+        // 12*Am[1] + 120*I
+#if defined(_MSC_VER)
+            for (i = 0; i < n; i++) {
+                Am[n2 + n*i + i] = CPLX_C(crealf(Am[n2 + n*i + i]) + b[0], cimagf(Am[n2 + n*i + i]));
+            }
+#else
+            for (i = 0; i < n; i++) { Am[n2 + n*i + i] += b[0]; }
+#endif
+
+    } else if (m == 5) {
+        // Am[3] and Am[4] are free
+        b[0] = 30240.0;
+        b[1] = 15120.0;
+        b[2] = 3360.0;
+        b[3] = 420.0;
+        b[4] = 30.0;
+
+        // U = Am[0] @ (b[5]*Am[2] + b[3]*Am[1] + b[1]*I_n)
+        // V = b[4]*Am[2] + b[2]*Am[1] + b[0]*I_n
+
+        // Am[4] = Am[1]
+        ccopy_(&n2, &Am[n2], &int1, &Am[4*n2], &int1);
+
+#if defined(_MSC_VER)
+        EXPM_C inter1, inter2;
+
+        for (i = 0; i < n; i++)
+        {
+            for (j = 0; j < n; j++)
+            {
+                inter1 = _FCmulcr(Am[n2 + n*i + j], b[3]);
+                inter2 = Am[2*n2 + n*i + j];
+                Am[4*n2 + n*i + j] =  CPLX_C(crealf(inter1) + crealf(inter2), cimagf(inter1)+ cimagf(inter2));
+                if (i == j) { Am[4*n2 + n*i + j] = CPLX_C(crealf(Am[4*n2 + n*i + j]) + b[1], cimagf(Am[4*n2 + n*i + j])); }
+            }
+        }
+#else
+        for (i = 0; i < n; i++)
+        {
+            for (j = 0; j < n; j++)
+            {
+                Am[4*n2 + n*i + j] = Am[2*n2 + n*i + j] + b[3]*Am[n2 + n*i + j];
+                if (i == j) { Am[4*n2 + n*i + j] += b[1]; }
+            }
+        }
+#endif
+
+        // Am[0] @ (1.0*Am[2] + 420*Am[1] + 15120*I)
+        cgemm_("N", "N", &n, &n, &n, &cdbl1, &Am[4*n2], &n, Am, &n, &cdbl0, &Am[3*n2], &n);
+
+        // V
+
+#if defined(_MSC_VER)
+        for (i = 0; i < n; i++)
+        {
+            for (j = 0; j < n; j++)
+            {
+                inter1 = _FCmulcr(Am[n2 + n*i + j], b[2]);
+                inter2 = _FCmulcr(Am[2*n2 + n*i + j], b[4]);
+                Am[n2 + n*i + j] = CPLX_C(crealf(inter1) + crealf(inter2), cimagf(inter1)+ cimagf(inter2));
+                if (i == j) { Am[n2 + n*i + j] = CPLX_C(crealf(Am[n2 + n*i + j]) + b[0], cimagf(Am[n2 + n*i + j])); }
+            }
+        }
+#else
+        for (i = 0; i < n; i++)
+        {
+            for (j = 0; j < n; j++)
+            {
+                Am[n2 + n*i + j] = b[4]*Am[2*n2 + n*i + j] + b[2]*Am[n2 + n*i + j];
+                if (i == j) { Am[n2 + n*i + j] = b[0]; }
+            }
+        }
+#endif
```

Comment:
@h-vetinari I wrote a lot of silly stuff like this with the hope that the MSVC compilers will go through. One job we have here with MSVC passed. I am wondering if you can also confirm it on your side with one case if you have the capacity. There are some precision based failures that I'll fix later so test results are not important but the build going through is my concern.

Reply:
I'll try to give this a spin tomorrow